### PR TITLE
Throw an exception if session cannot be started.

### LINKED ISSUE #17045: $_SESSION null when written to
### Description

Im upgrading our project from cakePHP 4.2 -> 4.3
On first write to Session.php $_SESSION is null. This warning is suppressed in [Session::write call to _overwrite](https://github.com/cakephp/cakephp/blob/4.3.8/src/Http/Session.php#L513), but because _overwrite() parameters are typed the call throws a type error
`protected function _overwrite(array &$old, array $new): void`

Session::write -> Session::start does not start our Session because we are using SSL (`session.use_cookies = true`) and our headers have been sent 
https://github.com/cakephp/cakephp/blob/4.3.8/src/Http/Session.php#L351

$_SESSION is sometimes allowed to be null. In the above situation $_SESSION is allowed to be null, but throws a typeError downstream. 
Why is the warning [here](https://github.com/cakephp/cakephp/blob/4.3.8/src/Http/Session.php#L513) suppressed?
Is $_SESSION allowed to be null in calls to Session::write? Where/how does $_SESSION get set?

### CakePHP Version

4.3.11

### PHP Version

8.0

### PR DESCRIPTION
**Pull request by ADmad**
This provides a better error message to developer if starting the session fails when trying to write to it.

Closes #17045.

### CODE REVIEW

**Comment by LordSimal** [comment]
Would look like this

image

for an example code like
```
trigger_error('test');
$this->getRequest()->getSession()->write('test', []);
```
I like 👍🏻

**Review by LordSimal — APPROVED** [review]

**Comment by ADmad** [comment]
The filename and line doesn't help much but it's still better than getting `TypeError` as mentioned in the related ticket :slightly_smiling_face:

**Comment by othercorey** [comment]
It's not required, but should we try to clip the filename to the project root path like we do with other error reporting?

**Comment by ADmad** [comment]
The would require access to the request instance and only way I can think of getting it here is through Router::getRequest() and I don't really like that.

File: src/Http/Session.php [hunk 1]
```diff
@@ -491,8 +502,18 @@ public function consume(string $name)
...
     {
-        if (!$this->started()) {
-            $this->start();
+        $started = $this->started() || $this->start();
+        if (!$started) {
+            $message = 'Could not start the session';
+            if ($this->headerSentInfo !== null) {
+                $message .= sprintf(
+                    ', headers already sent in file `%s` on line `%s`',
+                    $this->headerSentInfo['filename'],
```
Comment (garas) [inline src/Http/Session.php]:
```suggestion
                    Debugger::trimPath($this->headerSentInfo['filename']),
```
\+ `use Cake\Error\Debugger;`

**Pull request merged**